### PR TITLE
Fix the dumping of LclVar trees

### PR DESCRIPTION
**Pull request by briansull**
Introduced isLclFld flag to remove duplicated code
Corrected the dumping of struct promoted sub-fields

@dotnet/jit-contrib PTAL

### CODE REVIEW

File: src/jit/compiler.h [hunk 1]
```diff
@@ -275,9 +275,7 @@ class  LclVarDsc
     unsigned char       lvOverlappingFields :1;  // True when we have a struct with possibly overlapping fields
     unsigned char       lvContainsHoles     :1;  // True when we have a promoted struct that contains holes
     unsigned char       lvCustomLayout      :1;  // True when this struct has "CustomLayout"
-#if FEATURE_MULTIREG_ARGS_OR_RET
-    unsigned char       lvIsMultiRegArgOrRet:1; // Is this argument variable holding a value passed or returned in multiple registers?
-#endif
+    unsigned char       lvIsMultiRegArgOrRet:1;  // Is this a variable something that would be passed or returned in multiple registers?
```
Comment (CarolEidt) [inline src/jit/compiler.h]: Is this an unrelated change? Why would you remove the conditional compilation for this?

**Comment by CarolEidt** [comment]
Could you explain what was wrong with the dumping of the fields? Was it just the spacing?

**Comment by briansull** [comment]
No it wasn't just the spacing.
The case of struct promoted local fields were also handled 
 Previously  it would also print (last use) twice and some of the promoted local field info was different between the GT_LCL_VAR anf GT_LCL_FLD

**Comment by briansull** [comment]
I am going to with draw this PR as I only expect to have one file being changed.

Git did not behavior a I expected when I switched to a new branch.
